diff --git a/fsbl/plat/cv181x/bl2/bl2_opt.c b/fsbl/plat/cv181x/bl2/bl2_opt.c
index c73fa5574..013d5de2f 100644
--- a/fsbl/plat/cv181x/bl2/bl2_opt.c
+++ b/fsbl/plat/cv181x/bl2/bl2_opt.c
@@ -419,6 +419,21 @@ retry_from_flash:
 
        switch_rtc_mode_2nd_stage();
 
+       if (!(mmio_read_32(GPIO_BASE + 0x050) & (1 << 30))) {
+               NOTICE("=========================================\n");
+               NOTICE("||             Debug Mode              ||\n");
+               NOTICE("||                                     ||\n");
+               NOTICE("=========================================\n");
+
+               mmio_setbits_32(0x3003024, 1 << 6);     /* reset the small core */
+
+               /* pinmux was set for jtag by default */
+
+               while(1)
+                       ;
+       }
+
+
        if (monitor_entry) {
                NOTICE("Jump to monitor at 0x%lx.\n", monitor_entry);
                jump_to_monitor(monitor_entry, loader_2nd_entry);
