diff --git a/fsbl/plat/cv181x/bl2/bl2_main.c b/fsbl/plat/cv181x/bl2/bl2_main.c
index a4f1aec18..958bd9895 100644
--- a/fsbl/plat/cv181x/bl2/bl2_main.c
+++ b/fsbl/plat/cv181x/bl2/bl2_main.c
@@ -9,6 +9,7 @@
 #ifdef RTOS_ENABLE_FREERTOS
 int init_comm_info(int ret)
 {
+#if 0
        struct transfer_config_t *transfer_config = (struct transfer_config_t *)MAILBOX_FIELD;
        struct transfer_config_t transfer_config_s;
        unsigned char *ptr = (unsigned char *)&transfer_config_s;
@@ -34,7 +35,7 @@ int init_comm_info(int ret)
 
        transfer_config_s.checksum = checksum;
        memcpy(transfer_config, &transfer_config_s, sizeof(struct transfer_config_t));
-
+#endif
        return ret;
 }
 #endif
